common/prove/witness_helper.go

```diff
@@ -203,15 +203,11 @@ func (w *WitnessHelper) constructAccountWitness(
 		}
 		// it means this is a registerZNS tx
 		if proverAccounts == nil {
-			if accountKey != int64(len(*w.assetTrees)) {
+			if accountKey != w.assetTrees.Size() {
```

Comment:
This will not stand any longer, `accountKey` is the account index, it can be huge, while we wont allow size of `assetTrees` grow to huge `accountKey`.